extensions-api/src/main/scala/pl/touk/nussknacker/security/AuthCredentials.scala

```diff
@@ -1,3 +1,85 @@
 package pl.touk.nussknacker.security
 
-final case class AuthCredentials(value: String) extends AnyVal
+import java.nio.charset.StandardCharsets.UTF_8
+import java.util.Base64
+import javax.crypto.{Cipher, KeyGenerator, SecretKey}
+import scala.util.{Failure, Success, Try}
+
+sealed trait AuthCredentials
+
+object AuthCredentials {
+
+  val anonymousUsername     = "anonymous"
+  private val anonymousPass = "anonymous"
+
+  final case class PassedAuthCredentials(value: String) extends AuthCredentials
+  case object AnonymousAccess                           extends AuthCredentials
+
+  def fromStringToPassedAuthCredentials(value: String): AuthCredentials = {
```

Comment:
Shouldn't this and other stuff from here be in `security` module? Do we want to expose all this on api level? My understanding is we are telling users that they can use it and we have to communicate changes of this code then.

Reply:
i think, it could be hidden somehow, but at the moment the whole `AnonymousAccess` trait is exposed as part of the API. The `AuthCredentials` ADT is used there so, it'd require a non-trivial refactor.